### ISSUE TITLE
Set default globalsearch lookuptype

### ISSUE THREAD

**Issue by PieterL75** [netbox-community/netbox#17138]
### NetBox version

v4.0.8

### Feature type

Data model extension

### Proposed functionality

the default globalsearch lookuptype is set to partial match.
a user should be able to select its own preference for the default value.
another option, is that the last used lookuptype is saved for the user.

### Use case

as a user I prefer to have the 'regex' as a default search option. currently, I have to search, change the lookuptype, search again.

### Database changes

An extra selector field is required on the user profile.

**Comment by jeremystretch** [comment]
This seems like it would result in a lot of confusion and frustration. Initializing the form with a predictable value every time is more user-friendly.

**Comment by PieterL75** [comment]
I don't see how this can cause confusion. The setting should be retained per user. Each user than has the experience he choose.
I always change the search to regex, as looking for devices is easier like that (ex: server\d*\.customername)

**Comment by jeremystretch** [comment]
> I don't see how this can cause confusion. 

1. User changes the search type to "exact" and searches for something.
2. User does something else.
3. User returns and search for something entirely unrelated, but gets no results.
4. User is confused because results were expected.
5. User realizes the "exact" filter was still applied from an unrelated search, which was not intended.

**Comment by PieterL75** [comment]
That is not the behavior I propose. (But indeed an optional behavior, so let's ignore that one) 

The FR is too allow the user to set the default lookup type for him. This will give him a consistent experience.

**Comment by jeremystretch** [comment]
It's the same problem though, really. Once a default has been set for some temporary purpose, it will likely be forgotten.

**Comment by PieterL75** [comment]
I can't imagine one sets the default and forget about it. It's the main used feature of NetBox. All I do is search, change the lookup, search again...

Maybe get some more voices and poll on slack?

**Comment by tobikris** [comment]
What do you think of the following compromise?

* No setting is added for this.
* The URL to the search with regex matching selected is "exposed" to be able to create browser bookmarks.

I think this limits user confusion to the bare minimum as you need to click on your own bookmark and then ignore what is selected in the form.

**Comment by lgerhold** [comment]
I completely agree with the concerns raised in this issue. The current default setting for global search in NetBox is "partial match," which often leads to less relevant results, especially when searching for exact site names. Having to manually switch the lookup type to "exact match" for precise results is indeed cumbersome.

Implementing a feature that allows users to set their preferred default search behavior, or at least remembers the last used lookup type, would significantly enhance the user experience. This adjustment would streamline workflows by reducing repetitive actions and ensuring that search results align more closely with user expectations.

This enhancement aligns with previous discussions and issues where the need for more intuitive and user-friendly search functionalities has been highlighted.

**Comment by fggec** [comment]
I see both sides of that discussion. It would be nice to set the default search behavior for myself. On the other side i understand that for most standard users this could be confusing. 

My approach would be: If we can not have this feature requested maybe it is possible to change the partial match behavior to set the match which is nearest to the query (often the exact match) is shown on first position.

**Comment by PieterL75** [comment]
Isn't a regex the same as a partial match ?
if you are used to search for 'router0', the regex will still return the same results.
Except that if the regex is the default, you have the extra power to do 'router0*.site1' to return only the routers wit suffix 'site1'

**Comment by PieterL75** [comment]
There are already user-defined field.. Like how tables can be modified, the dashboard.. 
So why not go further on that personalization approach ? 
I prefer 'regex match' on al my searches as default, @lgerhold prefers 'exact'. and others might prefer something else.
It is more a matter of convenience than 'standards'. 
And if you (as a netbox admin) don't like that the option can be changed, then we can throw in an extra config var that enables/disabled this customization
Also, if the standard is 'regex' for everybody as of the start (of that netbox instance), then everybody is used to that way of searching.

**Comment by jeremystretch** [comment]
Given that it's already possible to searching using a regex under the global search page, this doesn't seem to be worth the effort and risk of user confusion.

**Comment by PieterL75** [comment]
I don't know, but this is a strange conclusion. 
Yes, I can select regex, but only after I searched. 

As a user, I want to control what my preferred search option is. 

Please

**Comment by peterbaumert** [comment]
I don't get either why many times concerns from users are put down like this. If he made up his mind its not discussion anymore sadly.

**Comment by jeremystretch** [comment]
I did not "put down" anyone's concerns. The proposed functionality was duly considered and discussed just like the literal thousands of other feature requests submitted to this project. You don't have to agree with my decision, but it is absolutely unfair to characterize it as dismissive just because you're unhappy with it.
